tsconfig.build.json

```diff
@@ -4,6 +4,8 @@
     "declaration": true,
     "module": "commonjs",
     "moduleResolution": "node",
+    "isolatedModules": true,
+    "noEmit": true,
```

Comment:
is there any reason to have this separate tsconfig file, since with this PR, the only thing `tsc` is used for is typechecking?

Reply:
re: tsconfig file and `prebuild` command.. I think I'll tackle these separately in cleaning up our `package.json` file